packages/dapp-svelte-wallet/api/src/wallet.js

```diff
@@ -49,167 +53,169 @@ export function buildRootObject(_vatPowers) {
           },
-          error(offer, reason) {
-            httpSend(
-              {
-                type: 'walletOfferResult',
-                data: {
-                  id: offer.id,
-                  dappContext: offer.dappContext,
-                  error: `${(reason && reason.stack) || reason}`,
-                },
-              },
-              [meta.channelHandle],
-            );
+          finish(offers) {
+            updater.finish(offers.filter(filter));
           },
-          handled(offer) {
-            if (handled) {
-              return;
-            }
-            handled = true;
-            httpSend(
-              {
-                type: 'walletOfferHandled',
-                data: offer.id,
-              },
-              [meta.channelHandle],
-            );
+          fail(e) {
+            updater.fail(e);
           },
         });
-        return {
-          type: 'walletOfferAdded',
-          data: await wallet.addOffer(
-            { ...data, actions },
-            { ...meta, dappOrigin },
-          ),
-        };
-      }
-      case 'walletDeclineOffer': {
-        return {
-          type: 'walletDeclineOfferResponse',
-          data: await wallet.declineOffer(data),
-        };
-      }
-      case 'walletCancelOffer': {
-        return {
-          type: 'walletCancelOfferResponse',
-          data: await wallet.cancelOffer(data),
-        };
-      }
-      case 'walletAcceptOffer': {
-        const { outcome } = await wallet.acceptOffer(data);
-        // We return the outcome only if it is a string. Otherwise we
-        // return a default message.
-        const result =
-          typeof outcome === 'string'
-            ? { outcome }
-            : { outcome: 'Offer was made.' };
-        return {
-          type: 'walletAcceptOfferResponse',
-          data: result,
-        };
-      }
+        return notifier;
+      },
+      async getDepositFacetId(brandBoardId) {
+        await approve();
+        return walletAdmin.getDepositFacetId(brandBoardId);
+      },
+      async suggestIssuer(petname, boardId) {
+        await approve();
+        return walletAdmin.suggestIssuer(petname, boardId, dappOrigin);
+      },
+      async suggestInstallation(petname, boardId) {
+        await approve();
+        return walletAdmin.suggestInstallation(petname, boardId, dappOrigin);
+      },
+      async suggestInstance(petname, boardId) {
+        await approve();
+        return walletAdmin.suggestInstance(petname, boardId, dappOrigin);
+      },
+    };
+    return harden(bridge);
+  };
 
-      case 'walletGetOffers':
-      case 'walletGetOfferDescriptions': {
-        const result = await wallet.getOffers({ origin: dappOrigin });
-        return {
-          type: 'walletOfferDescriptions',
-          data: result,
+  /**
+   * This bridge doesn't wait for approvals before acting.
+   *
+   * @type {WalletBridge}
+   */
+  const preapprovedBridge = {
```

Comment:
If possible, I would like more comments about when and where things are expected to be used. Right now I don't think there's anything written down on why a preapprovedBridge exists or why it is needed, but that should be in the comments here.